guardian/grid-feeds
# GuCDK and RiffRaff Deployment

## What does this change?
This creates the Github Actions CI, RiffRaff deploy and GuCDK infrastructure for the AP API Feed service.

It additionally adds some more error handling to the code and reading/writing to DynamoDB to store the next page in the feed.

There is still some outstanding work on this project which will be added in subsequent PRs:

- [ ] improve memory management to better guard against OutOfMemoryError exceptions
- [ ] add success metrics and an alert to tell us if we have not uploaded any images (in an hour)
- [ ] improve long polling logic to restart thread if the process is interrupted

## How to test
Check the app in CODE. It is currently deployed to publish live AP images to the grid testing environment, so you should see images successfully coming through here: https://media.test.dev-gutools.co.uk/search?query=uploader:ap&nonFree=true. You can also check the application logs in Kibana: https://logs.gutools.co.uk/s/editorial-tools/goto/d5750450-b9ad-11ed-9f42-a9a60d1fdfa3
